src/egon/data/processing/power2gas/__init__.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+"""
+Definition of the power-to-gas installations
+"""
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import shapely.ops as sp_ops
+
+from shapely import geometry
+from geoalchemy2.types import Geometry
+from scipy.spatial import cKDTree
+from pyproj import Transformer
+from egon.data import db
+from egon.data.importing.gas_grid import next_id
+
+
+def insert_power2gas():
+    sql_AC = """SELECT bus_id, geom 
+                FROM grid.egon_pf_hv_bus 
+                WHERE carrier = 'AC';"""
+    sql_gas = """SELECT bus_id, version, scn_name, geom 
+                FROM grid.egon_pf_hv_bus 
+                WHERE carrier = 'gas';""" 
+    
+    # Connect to local database
+    engine = db.engine()
+    
+    # Select next id value
+    new_id = next_id('link')
+        
+    gdf_AC = gpd.read_postgis(sql_AC, engine, crs=4326)
+    gdf_gas = gpd.read_postgis(sql_gas, engine, crs=4326)
+
+    
+    if gdf_AC.size == 0:
+        print("WARNING: No data returned in grid.egon_pf_hv_bus WHERE carrier = 'AC'")
+    elif gdf_gas.size == 0: 
+        print("WARNING: No data returned in grid.egon_pf_hv_bus WHERE carrier = 'gas'")

```

Comment:
The function [db.select_geodataframe()](https://github.com/openego/eGon-data/blob/255d9653ab8b24c4a3c63fb9c06d97ef3a2262c8/src/egon/data/db.py#L164-L192) includes this check. If you use this, you don't have to do it manually.

Reply:
Thank you for the advise. I've changed it.